pkg/registry/core/persistentvolume/strategy.go

```diff
@@ -139,6 +139,8 @@ func (persistentvolumeStatusStrategy) PrepareForUpdate(ctx context.Context, obj,
 	newPv := obj.(*api.PersistentVolume)
 	oldPv := old.(*api.PersistentVolume)
 	newPv.Spec = oldPv.Spec
+
```

Comment:
An alternative approach to update the `LastPhaseTransitionTime` field would be to update it here

```
if old.status.phase != new.status.phase {
   new.status.lastPhaseTransitionTime = now
}
```

So when *user* or some other controller modifies the phase, the timestamp gets updated too. I am leaving it here as an alternative.

Reply:
Good point, I think we have not considered that in the design initially. Let's see if we get more feedback or find a use case for allowing phase updates from other components or users, we can change this later. To me it makes more sense now to leave this field for pv controller only.

Reply:
Done - moved to strategy.